prboom2/src/gl_main.c

```diff
@@ -823,14 +824,15 @@ void gld_DrawWeapon(int weaponlump, vissprite_t *vis, int lightlevel)
     glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
   }
   glColor3f(1.0f,1.0f,1.0f);
-  gld_EndFuzz();
+  if (/*(viewplayer->mo->flags & MF_SHADOW) && */!vis->colormap)
```

Comment:
Bug fix?

Reply:
Without this check to mirror the one for `StartFuzz`, you can get unbalanced push/pop of the shader stack.